cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsReporterSampler.java

```diff
@@ -155,6 +144,27 @@ private boolean consumptionDone(Map<TopicPartition, Long> endOffsets) {
     return true;
   }
 
+  /**
+   * Ensure that the {@link #_metricConsumer} is assigned to the latest partitions of the {@link #_metricReporterTopic}.
+   * This enables metrics reporter sampler to handle dynamic partition size increases in {@link #_metricReporterTopic}.
+   *
+   * @return True if the set of partitions currently assigned to this consumer is empty, false otherwise.
+   */
+  private boolean refreshPartitionAssignment() {
+    _currentPartitionAssignment = new HashSet<>();
+    for (PartitionInfo partitionInfo : _metricConsumer.partitionsFor(_metricReporterTopic)) {
+      _currentPartitionAssignment.add(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
+    }
+
+    if (_currentPartitionAssignment.isEmpty()) {
+      LOG.error("The set of partitions currently assigned to the metric consumer is empty.");
+      return true;
+    }
+
+    _metricConsumer.assign(_currentPartitionAssignment);
```

Comment:
I think a better approach maybe check `_currentPartitionAssignment` is changed or not,if is changed, then call `assign()`(instead of alway call `assign()` in this method).